emigration.lua

```diff
@@ -4,13 +4,14 @@
 local GLOBAL_KEY = 'emigration' -- used for state change hooks and persistence
 
 enabled = enabled or false
+last_cycle_tick = last_cycle_tick or 0
```

Comment:
now that we have more than just `enabled` here, I suggest making this clearer by putting the state in a table.

```lua
local function get_default_state()
    return {enabled=false, last_cycle_tick=0}
end

state = state or get_default_state()
```